### ISSUE TITLE
Having issue getting SC2 version 4.10.0 on windows 10

### ISSUE THREAD

**Issue by laughinggaschambers** [opendilab/DI-star#33]
Following the windows software test installation, pressing the activate 4.10.0 button the replay does not play or download version 4.10.0
![signal-2023-04-09-222529_002](https://user-images.githubusercontent.com/114764161/230781633-6b98c6ec-71cf-42fb-9012-e285d324312b.png)

**Comment by upia99** [comment]
thanks for reporting, I replace the broken replay with a new one, check the new commit. You can download the new windows package or only replace the replay located in /distar/data/replays/replay_4.10.0.SC2Replay

**Comment by laughinggaschambers** [comment]
I still having the same issues, seems to be a problem with version 4.10.0 in general, I've attempted to download replays of this version from other websites too and had the same issue.

**Comment by ty359** [comment]
![image](https://user-images.githubusercontent.com/9530794/232050117-3b311e51-e3f4-4949-bff4-d72508245945.png)
same issue

**Comment by upia99** [comment]
I did solve the problem by replacing the replay. try this:
1. update StarCraft 2 to the newest version on Battle.net
2. remove the old files of version 4.10.0 by deleting the folder: <your SC2 installation path>\StarCraft II\Versions\Base75689
3. then double click the replay [(download path)](https://drive.google.com/file/d/1lNkSVIaaXXBF7agqeX9H2_-5G-Zjbp-J/view?usp=share_link) and wait 4.10.0 to download, close the game if the replay start to play.
4. all done, play your game on DI-star windows software.

**Comment by shuod** [comment]
@upia99 
so after your step3, where do I put the "replay_4.10.0.SC2Replay"

**Then,** in the window of DI-star, do I still choose activate 4.10 button there? but once I click on it, it will start SC2.exe and then error like @laughinggaschambers reported will show up "mod or map no longer available to download"

**Or,** if I click on the start button in the DI-star window, the python terminal will report "Unknown game version: 4.10.0"
`warning! cuda is not activate, this will cause significant agent performance degradation!
{'var1': 'default', 'common': {'type': 'play'}, 'agent': {'z_path': '7map_filter_spine.json', 'show_Z': True, 'zero_z_exceed_loop': True, 'extra_units': True, 'fake_reward_prob': 0.8}, 'feature': {'zero_z_value': 1.0, 'zergling_num': 8}, 'actor': {'job_type': 'eval_test', 'gpu_batch_inference': False, 'env_num': 1, 'episode_num': 1, 'print_freq': 1000, 'traj_len': 2, 'use_cuda': False, 'fake_model': False, 'player_ids': ['model1'], 'agents': {'model1': 'default', 'model2': 'default'}, 'model_paths': {'model1': '../distar/bin\\rl_model.pth', 'model2': 'default'}, 'teacher_player_ids': ['haha', 'haha'], 'teacher_model_paths': {'haha': 'haha.pth.tar'}}, 'env': {'map_name': 'random', 'player_ids': ['rl_model', 'human'], 'races': ['zerg', 'zerg'], 'map_size_resolutions': [True, True], 'minimap_resolutions': [[160, 152], [160, 152]], 'realtime': True, 'replay_dir': '../replays', 'game_steps_per_episode': 100000, 'update_both_obs': False, 'version': '4.10.0'}}
[MAIN LOOP ERROR] Unknown game version: 4.10.0. Known versions: ['latest'].
  File "distar\actor\actor.py", line 328, in run
    else:
  File "distar\actor\actor.py", line 115, in _inference_loop
    self._env = SC2Env(mergerd_whole_cfg)
  File "distar\envs\env.py", line 135, in __init__
    self._run_config = run_configs.get(version=self._version)
  File "distar\pysc2\run_configs\__init__.py", line 38, in get
    return max(configs.values(), key=lambda c: c.priority())(version=version)
  File "distar\pysc2\run_configs\platforms.py", line 153, in __init__
    version=version, cwd="Support64")
  File "distar\pysc2\run_configs\platforms.py", line 95, in __init__
    data_dir=base_dir, tmp_dir=None, version=version, cwd=cwd, env=env)
  File "distar\pysc2\run_configs\lib.py", line 128, in __init__
    self.version = self._get_version(version)
  File "distar\pysc2\run_configs\lib.py", line 239, in _get_version
    versions = self.get_versions(containing=game_version)
  File "distar\pysc2\run_configs\platforms.py", line 141, in get_versions
    containing, sorted(ret.keys())))`

**Comment by shuod** [comment]
@upia99 
ok, i put the downloaded 
replay_4.10.0.SC2Replay from your step 3 into the folder distar\data\replays

I download the DI-star package from your link from Tencent Cloud drive.

still same errors.

**Comment by upia99** [comment]
@shuod Did you try directly double click the replay? If you get the same error, try to reinstall SC2, I have tested it with 2 PC. I guess the only difference is I install SC2 for the new PC.

**Comment by puckikk1202** [comment]
`[MAIN LOOP ERROR] Unknown game version: 4.10.0. Known versions: ['latest'].`
I have also followed your 3 steps but it is still not working. Is it possible that version 4.10.0 is invalid after the recent new game patch was updated from the official?

**Comment by GUOJINHAO-XIA** [comment]
I encountered the same problem. Now it is January 13th, 2025, and the version of starcraft is 5.0.14. I downloaded the Starcraft client, and distar, I clicked to activate 4.10. After downloading, I started the client. Like the subject's problem, I reported an error, and I couldn't replay the 4.10 file. Why?

**Comment by GUOJINHAO-XIA** [comment]
> @upia99 那么完成第 3 步后，我应该将“replay_4.10.0.SC2Replay”放在哪里？
> 
> **现在**在DI-star的窗口中，我是否仍然选择激活4.10按钮？但是一旦我点击它，它就会启动SC2.exe，然后出现类似以下错误@laughinggaschambers报告将显示“模组或地图不再提供下载”
> 
> **或者，**如果我点击DI-star窗口中的开始按钮，python终端将报告“未知的游戏版本：4.10.0” `warning! cuda is not activate, this will cause significant agent performance degradation! {'var1': 'default', 'common': {'type': 'play'}, 'agent': {'z_path': '7map_filter_spine.json', 'show_Z': True, 'zero_z_exceed_loop': True, 'extra_units': True, 'fake_reward_prob': 0.8}, 'feature': {'zero_z_value': 1.0, 'zergling_num': 8}, 'actor': {'job_type': 'eval_test', 'gpu_batch_inference': False, 'env_num': 1, 'episode_num': 1, 'print_freq': 1000, 'traj_len': 2, 'use_cuda': False, 'fake_model': False, 'player_ids': ['model1'], 'agents': {'model1': 'default', 'model2': 'default'}, 'model_paths': {'model1': '../distar/bin\\rl_model.pth', 'model2': 'default'}, 'teacher_player_ids': ['haha', 'haha'], 'teacher_model_paths': {'haha': 'haha.pth.tar'}}, 'env': {'map_name': 'random', 'player_ids': ['rl_model', 'human'], 'races': ['zerg', 'zerg'], 'map_size_resolutions': [True, True], 'minimap_resolutions': [[160, 152], [160, 152]], 'realtime': True, 'replay_dir': '../replays', 'game_steps_per_episode': 100000, 'update_both_obs': False, 'version': '4.10.0'}} [MAIN LOOP ERROR] Unknown game version: 4.10.0. Known versions: ['latest']. File "distar\actor\actor.py", line 328, in run else: File "distar\actor\actor.py", line 115, in _inference_loop self._env = SC2Env(mergerd_whole_cfg) File "distar\envs\env.py", line 135, in __init__ self._run_config = run_configs.get(version=self._version) File "distar\pysc2\run_configs\__init__.py", line 38, in get return max(configs.values(), key=lambda c: c.priority())(version=version) File "distar\pysc2\run_configs\platforms.py", line 153, in __init__ version=version, cwd="Support64") File "distar\pysc2\run_configs\platforms.py", line 95, in __init__ data_dir=base_dir, tmp_dir=None, version=version, cwd=cwd, env=env) File "distar\pysc2\run_configs\lib.py", line 128, in __init__ self.version = self._get_version(version) File "distar\pysc2\run_configs\lib.py", line 239, in _get_version versions = self.get_versions(containing=game_version) File "distar\pysc2\run_configs\platforms.py", line 141, in get_versions containing, sorted(ret.keys())))`

Have you solved it? I have the same problem as you.

**Comment by leolui2004** [comment]
I got the same error
`[MAIN LOOP ERROR] Unknown game version: 4.10.0. Known versions: ['latest']`
when I run the command `python -m distar.bin.play` after installation.

As I did not installed 4.10.0 thus I thought there is any thing wrongly hard-coded in the file.
**And I found that there is a file user_config.yaml in distar/bin, which stated the version is 4.10.0.**
This config file is called when starting the game.
**After I changed the setting to 5.0.14, and added the version info to distar/pysc2/run_configs/lib.py, the problem solved.**
Hope it helps ;)
